eng/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -43,32 +44,93 @@ function GenerateMatrix([MatrixConfig]$config, [string]$selectFromMatrixType)
         [Array]$matrix = ProcessIncludes $matrix $config.include $config.displayNamesLookup
     }
 
+    [Array]$matrix = FilterMatrixDisplayName $matrix $displayNameFilter
+    [Array]$matrix = FilterMatrix $matrix $filters
+    return $matrix
+}
+
+function FilterMatrixDisplayName([array]$matrix, [string]$filter) {
+    return $matrix | ForEach-Object {
+        if ($_.Name -match $filter) {
+            return $_
+        }
+    }
+}
+
+# Filters take the format of key=valueregex,key2=valueregex2
+function FilterMatrix([array]$matrix, [array]$filters) {
+    $matrix = $matrix | ForEach-Object {
+        if (MatchesFilters $_ $filters) {
+            return $_
+        }
+    }
     return $matrix
 }
 
+function MatchesFilters([hashtable]$entry, [array]$filters) {
+    $nonMatching = $filters | Where-Object { $null -ne $_ } | ForEach-Object {
```

Comment:
Might I suggest that this function might be easier to understand if you just a standard foreach loop instead of doing weirdness with the PS pipeline syntax (i.e. returning nothing in some cases and false in others).

Reply:
You may! Yeah it would be better that way.